DOCUMENTATION.md

```diff
@@ -29,6 +30,12 @@ This module implements all models for Resin Python SDK.
 ## Application
 
 This class implements application model for Resin Python SDK.
+
+Due to API changes, the returned Application object schema has changed. Here are the formats of the old and new returned objects.
+
+The old returned object's properties: `__metadata, actor, app_name, application, commit, device_type, git_repository, id, support_expiry_date, user, version, should_track_latest_release`.
+
+The new returned object's properties (since Python SDK v1.7.0): `__metadata, actor, app_name, commit, device_type, git_repository, id, is_accessible_by_support_until__date, should_track_latest_release, user, version, depends_on__application`.
```

Comment:
This is a breaking change. It looks like we rereleased it as 2.0.0, which is the right thing to do :+1:, but we should also stop people installing 1.7.0, since it'll break their code. Can we not unrelease/deprecate `1.7.0` and change these explanations to reference `2.0.0`?

Reply:
Hmmm, I suppose in hindsight the "Upgrade Pine endpoint to Resin API v3." of 1.7.0 (see [CHANGELOG](https://github.com/resin-io/resin-sdk-python/blob/master/CHANGELOG.md) ) in #53 should have necessitated a 'major' version-bump rather than a 'minor' version-bump? (it even says `includes many breaking changes` in the PR description)

And in contrast, perhaps adding Python3 support would have been a 'minor' rather than a 'major' change? I guess it depends how strictly you want to apply semver?

Reply:
> perhaps adding Python3 support would have been a 'minor' rather than a 'major' change?

I assumed that fixing this was what the 2.0.0 change was for, not adding python 3! Adding python 3 support is almost certainly not a breaking change, you're right.

> I guess it depends how strictly you want to apply semver?

Imo, this isn't a particularly strict interpretation of semver: the currently released version of 1.7.0 is unexpectedly breaking user's code in the real world with no warning (#57), because it wasn't marked as major. There are other cases where it's technically breaking but probably doesn't affect anybody where  it can be pretty debatable, but I think this one is pretty clear cut.

On fixing it though, my suggestion was wrong, the semver spec actually says:

> What do I do if I accidentally release a backwards incompatible change as a minor version?
>
> As soon as you realize that you’ve broken the Semantic Versioning spec, fix the problem and release a new minor version that corrects the problem and restores backwards compatibility. Even under this circumstance, it is unacceptable to modify versioned releases. If it’s appropriate, document the offending version and inform your users of the problem so that they are aware of the offending version.

I.e. we should not remove 1.7.0, but instead ship a 1.8.0 that reverts the API bump just for the 1.x line, to help users easily avoid the breakage (and then continue as normal from 2.0.0). As far as I can tell there's no way to mark a package version as broken or add a warning to it in pypi, but that's what I'd do if this were npm.

@nghiant2710 we should have a chat about the python SDK this week and sort all this out properly. Are you free at 11am UTC tomorrow? (I think that's 6pm your time)

Reply:
And of course this also all links (a bit) to #58

Reply:
@pimterry, sure 11.am UTC works for me.